### PR TITLE
Don't use new Buffer()

### PR DESCRIPTION
**Pull request by RyanZim**
Please review very carefully, as this lib does not have functioning tests.

### CODE REVIEW

File: lib/util/buffer.js [hunk 1]
```diff
@@ -109,7 +109,7 @@ module.exports = {
     bytes.push((integer >> 16) & 0xff);
     bytes.push((integer >> 8) & 0xff);
     bytes.push(integer & 0xff);
-    return new Buffer(bytes);
+    return Buffer.alloc(bytes);
```
Comment (mvayngrib) [inline lib/util/buffer.js]:
```suggestion
    return Buffer.from(bytes);
```

Comment (RyanZim) [reply lib/util/buffer.js]: This is why we do this review thingy.

**Review by mvayngrib — APPROVED** [review]
utACK

**Review by kklash — APPROVED** [review]
utACK

**Review by roccomuso — APPROVED** [review]
tACK on desktop.
- got same address
- send/receive tx

**Review by ChALkeR — DISMISSED** [review]
This is not complete.

```js
lib/transaction/input/input.js:    prevTxId = new buffer.Buffer(params.prevTxId, 'hex');
lib/transaction/input/input.js:    this._scriptBuffer = new buffer.Buffer(script, 'hex');
lib/transaction/input/input.js:    this._scriptBuffer = new buffer.Buffer(script);
lib/transaction/transaction.js:  this.fromBuffer(new buffer.Buffer(string, 'hex'));
lib/transaction/sighash.js:        atoms: BN.fromBuffer(new buffer.Buffer(BITS_64_ON, 'hex')),
lib/transaction/output.js:        script = new buffer.Buffer(args.script, 'hex');
lib/transaction/output.js:    obj.script = new buffer.Buffer([]);
lib/util/buffer.js:    var result = new buffer.Buffer(bytes);
lib/util/buffer.js:    return new buffer.Buffer([integer & 0xff]);
lib/util/buffer.js:    var ret = new buffer.Buffer(param.length);
lib/util/buffer.js:    return new buffer.Buffer(string, 'hex');
lib/encoding/base58check.js:    data = new buffer.Buffer(Base58.decode(data));
lib/encoding/base58check.js:    checksum = new buffer.Buffer(Base58.decode(checksum));
lib/hdprivatekey.js:    data = BufferUtil.concat([new buffer.Buffer([0]), this.privateKey.toBuffer(), indexBuffer]);
lib/hdprivatekey.js:  var hash = Hash.sha512hmac(hexa, new buffer.Buffer('Bitcoin seed'));
lib/script/script.js:  return new Script(new buffer.Buffer(str, 'hex'));
lib/script/script.js:    return new Script(new buffer.Buffer(str, 'hex'));
```

In all those files:
```js
lib/transaction/input/input.js:var buffer = require('buffer');
lib/transaction/transaction.js:var buffer = require('buffer');
lib/transaction/sighash.js:var buffer = require('buffer');
lib/transaction/output.js:var buffer = require('buffer');
lib/util/buffer.js:var buffer = require('buffer');
lib/encoding/base58.js:var buffer = require('buffer');
lib/encoding/base58check.js:var buffer = require('buffer');
lib/hdprivatekey.js:var buffer = require('buffer');
lib/script/script.js:var buffer = require('buffer');
```

**Review by RyanZim — COMMENTED** [review]
utACK

**Review by kklash — APPROVED** [review]
utACK

**Comment by roccomuso** [comment]
tACK

- [x] send/receive

**Review by roccomuso — APPROVED** [review]

**Review by mvayngrib — APPROVED** [review]
utACK

**Comment by sonaye** [comment]
What's the motivation behind these changes, did something break? I'd rather we spend some time on https://github.com/bitcoinjs/bitcoinjs-lib/issues/513 instead

**Comment by RyanZim** [comment]
@sonaye This is part of a larger project to remove all usage of `new Buffer` in Exodus desktop. I'm with you, long-term, I'd prefer that solution as well (hate dealing with this lib).

**Comment by RyanZim** [comment]
Rebased to fix merge conflicts; @ChALkeR got a few tests running in https://github.com/ExodusMovement/dcrcore-lib/pull/8; they pass.

**Pull request merged**